### PR TITLE
feat: added admin_key to nodeCreate, nodeUpdate and unit tests

### LINKED ISSUE #13981: Add admin_key to nodeCreate/Update and unit tests

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

Added admin_key support in nodeCreate/Update and unit test.
All handlers unit tests are 100% methods and 100% lines covered.
Also added updateAccountIdAllowed flag check.

**Related issue(s)**:

Fixes #13981

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/AccountsConfig.java [hunk 1]
```diff
@@ -33,7 +33,6 @@ public record AccountsConfig(
         @ConfigProperty(defaultValue = "50") @NetworkProperty long systemAdmin,
         @ConfigProperty(defaultValue = "59") @NetworkProperty long systemDeleteAdmin,
         @ConfigProperty(defaultValue = "60") @NetworkProperty long systemUndeleteAdmin,
-        @ConfigProperty(defaultValue = "53") @NetworkProperty long nodeAdmin,
```
[the remark this answers is not in the document]
Comment (iwsimon) [reply hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/AccountsConfig.java:36]: We have 55 defined here as addressBookAdmin. Used this value instead.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java [hunk 2]
```diff
@@ -58,19 +59,39 @@ public NodeUpdateHandler(@NonNull final AddressBookValidator addressBookValidato
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        requireNonNull(txn);
         final var op = txn.nodeUpdate();
         validateFalsePreCheck(op.nodeId() < 0, INVALID_NODE_ID);
-        if (op.hasAccountId()) {
-            final var accountId = validateAccountID(op.accountIdOrElse(AccountID.DEFAULT), INVALID_NODE_ACCOUNT_ID);
-            validateFalsePreCheck(!accountId.hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
-        }
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java:66]: Why is this removed from pureChecks ?

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java [hunk 3]
```diff
@@ -95,12 +116,16 @@ public void handle(@NonNull final HandleContext handleContext) {
             addressBookValidator.validateGossipEndpoint(op.gossipEndpoint(), nodeConfig);
         if (!op.serviceEndpoint().isEmpty())
             addressBookValidator.validateServiceEndpoint(op.serviceEndpoint(), nodeConfig);
+        if (op.hasAdminKey()) addressBookValidator.validateAdminKeyInHandle(handleContext, op.adminKeyOrThrow());
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java:119]: We dont need `InHandle` in the method name

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java [hunk 4]
```diff
@@ -105,14 +124,58 @@ public void validateGossipEndpoint(
...
+     * Validates the admin key.
+     *
+     * @param key The key to validate
+     * @throws PreCheckException if the key is invalid
+     */
+    public void validateAdminKey(@Nullable Key key) throws PreCheckException {
+        final var keyEmpty = isEmpty(key);
+        validateFalsePreCheck(key == null || keyEmpty, KEY_REQUIRED);
+        validateTruePreCheck(isValid(key), INVALID_ADMIN_KEY);
+    }
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java:146]: Why do we need a different method for this ? `isEmpty` is already checked in `isValid` . We could just use isValid and null without this method

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java [hunk 5]
```diff
@@ -105,14 +124,58 @@ public void validateGossipEndpoint(
...
+    public void validateAdminKeyInHandle(@NonNull final HandleContext handleContext, @NonNull final Key key) {
+        requireNonNull(handleContext);
+        requireNonNull(key);
+
+        try {
+            handleContext.attributeValidator().validateKey(key);
+        } catch (HandleException e) {
+            throw new HandleException(INVALID_ADMIN_KEY);
+        }
+    }
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java:162]: We should add validateKey that accepts response code in AttributeValidator

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java [hunk 6]
```diff
@@ -105,14 +124,58 @@ public void validateGossipEndpoint(
...
+
+    /**
+     * Validates the account ID.
+     *
+     * @param accountId The account ID to validate
+     * @throws PreCheckException if the account ID is invalid
+     */
+    public void validateAccountId(@Nullable AccountID accountId) throws PreCheckException {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java:170]: We have `validateAccountID` in spi `Validations` class. We can use that right ?

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java [hunk 7]
```diff
@@ -66,26 +66,30 @@ public NodeCreateHandler(@NonNull final AddressBookValidator addressBookValidato
...
-        validateFalsePreCheck(!accountId.hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
+        addressBookValidator.validateAccountId(op.accountId());
         validateFalsePreCheck(op.gossipEndpoint().isEmpty(), INVALID_GOSSIP_ENDPOINT);
         validateFalsePreCheck(op.serviceEndpoint().isEmpty(), INVALID_SERVICE_ENDPOINT);
         validateFalsePreCheck(
                 op.gossipCaCertificate().length() == 0
                         || op.gossipCaCertificate().equals(Bytes.EMPTY),
                 INVALID_GOSSIP_CA_CERTIFICATE);
+
+        final var adminKey = op.adminKey();
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java:79]: Cane we throw error when `!op.hasAdminKey()` here ? and we can directly use `isValid` check here

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java (hunk 2 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java:66]: Need to add config.updateAccountIdAllowed() check. New Requirements changed in #dynamic-addressbook.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java (hunk 6 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java:170]: Validations.validateAccountID() is called in this method. We also need to check alias is not valid. So added that check here.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java (hunk 3 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java]: Removed.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java (hunk 4 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java]: Removed null check. I wanted to return a different error code when key is empty and key is invalid. Added this method to take care of both.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java (hunk 5 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java]: Added com.hedera.node.app.spi.validation.AttributeValidator#validateKey(com.hedera.hapi.node.base.Key, com.hedera.hapi.node.base.ResponseCodeEnum), removed this method.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java (hunk 7 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java:79]: Used validateAdminKey() to take care both senarios.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java (hunk 6 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java:152]: Ah i see , we can move the alias check before `validateAccountID`

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM! Just one minor comment. Thanks @iwsimon

**Review by JeffreyDallas — APPROVED** [review]

**Review by rbarkerSL — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by rbarkerSL — APPROVED** [review]
